src/uct/ugni/base/ugni_md.h

```diff
@@ -56,4 +56,10 @@ extern pthread_mutex_t uct_ugni_global_lock;
  * @brief Helper function to list UGNI resources
  */
 uct_ugni_device_t * uct_ugni_device_by_name(const char *dev_name);
+
+static UCS_F_ALWAYS_INLINE gni_nic_device_t uct_ugni_find_gni_device_type(int dev_id)
```

Comment:
This is not a fast past operation, so I don't think we have to force inlining. 
I would suggest to go with a regular inline and let the compiler to decide.